### PR TITLE
Switch public API back to REST using Spring Cloud Gateway to proxy

### PR DESCRIPTION
**Pull request by itzg**
# Resolves

https://jira.rax.io/browse/SALUS-252

# What

As discussed in https://github.com/racker/salus-telemetry-api/pull/20, GraphQL became very awkward to work with for our APIs primarily due to the polymorphic nature of our monitor and event task configuration. This PR switches us (back) to a REST API.

**NOTE** you might find reviewing the final content easier than the diffs since a majority of the code was completely swapped out.

# How

Way down at the bottom of the Spring Cloud Gateway docs, I found they have a mechanism for building easily customizable proxy controllers: https://cloud.spring.io/spring-cloud-static/spring-cloud-gateway/2.1.0.RELEASE/multi/multi__building_a_simple_gateway_using_spring_mvc_or_webflux.html . Most of the controllers contain the very simple pattern of:
- locate the tenant via Spring Security (via our `UserService`)
- determine the backend URI to use
- proxy the call to the backend API

The exception to the pattern at this time is the monitor API. For now, the awareness of plugin specific configuration is modeled and validated by the public API controller. Then, the call to the backend monitor API is given the configuration JSON blob via the `content` field of the monitor. If others agree, we should probably shift that awareness down into the monitor management service. That way we can evolve that service and deploy it independent of the user facing public API.

# TODO

- [x] Adding unit tests for monitor to JSON config blob conversion
- [x] Update the Envoy to process the small alteration to the JSON config blob which now encodes a single plugin config and the `type` field is a part of the config object. https://github.com/racker/salus-telemetry-api/pull/22
- I might switch to the webflux flavor of the gateway since [there is presently an issue](https://github.com/spring-cloud/spring-cloud-gateway/issues/674) sometimes when the backend service responds with a 400. This was seen specifically when the `POST /api/eventTasks` was given an empty JSON object body. Switching would mainly require re-implementing the `ReposeHeaderFilter` in a [webflux way](https://docs.spring.io/spring-security/site/docs/current/reference/html5/#reactive-applications).

### CODE REVIEW

**Comment by GeorgeJahad** [comment]
I do agree it would be better to put the monitor config manipulation stuff in the monitor api.  the api gateway should just be doing security and routing and general cross api manipulation.

**Comment by itzg** [comment]
Created https://jira.rax.io/browse/SALUS-261 for follow-up shift of monitor config blob handling.

File: public/src/main/java/com/rackspace/salus/telemetry/api/web/EventTasksController.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@RestController
+@RequestMapping("/api/eventTasks")
+@SuppressWarnings("Duplicates") // due to repetitive proxy setup/calls
```
Comment (GeorgeJahad) [inline public/src/main/java/com/rackspace/salus/telemetry/api/web/EventTasksController.java]: Nice, I didn't know how to do that.

Comment (itzg) [reply public/src/main/java/com/rackspace/salus/telemetry/api/web/EventTasksController.java]: IntelliJ helped me with that :) I did an Option-Return on the grey swiggly lines and picked this:

image

**Comment by GeorgeJahad** [comment]
oops, i was looking at my local copy and didn't realize you committed more stuff.  i approved prematurely, sorry.  looking at the latest commits now

File: public/src/main/java/com/rackspace/salus/telemetry/api/services/MonitorConversionService.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+/**
+ * This service enables the public API to interface with the backend REST services for monitor management.
+ */
+@Service
+@Slf4j
+public class MonitorConversionService {
+
+  private final ObjectMapper objectMapper;
+  private final JsonNodeFactory nodeFactory;
```
Comment (GeorgeJahad) [inline public/src/main/java/com/rackspace/salus/telemetry/api/services/MonitorConversionService.java]: nit:  this doesn't seem used.

Comment (itzg) [reply public/src/main/java/com/rackspace/salus/telemetry/api/services/MonitorConversionService.java]: oops, yep, I'll remove that

**Pull request merged**